drivers/rtc/rtc_stm32.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2020 Paratronic
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Source file for the STM32 RTC driver
+ *
+ */
+
+
+#include <kernel.h>
+#include <soc.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/rtc.h>
+#include <sys/timeutil.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(rtc_stm32, LOG_LEVEL_DBG);
+
+#if defined(CONFIG_RTC_STM32_CLOCK_LSI)
+/* prescaler values for LSI @ 32 KHz */
+#define RTC_PREDIV_ASYNC 31
+#define RTC_PREDIV_SYNC  999
+#else /* CONFIG_COUNTER_RTC_STM32_CLOCK_LSE */
+/* prescaler values for LSI @ 32768 Hz */
+#define RTC_PREDIV_ASYNC 31
+#define RTC_PREDIV_SYNC  1023
+#endif
+
+static int rtc_stm32_settime(struct device *dev, const struct timespec *tp)
+{
+	int err = -1;
+	struct tm tm;
+	LL_RTC_TimeTypeDef timeDef;
+	LL_RTC_DateTypeDef dateDef;
+
+	LL_PWR_EnableBkUpAccess();
```

Comment:
The path between LL_PWR_EnableBkUpAccess and LL_PWR_DisableBkUpAccess is a critical section. IMHO it should be protected by a e.g. semaphore preventing threads to access at the same time.

Reply:
Now, I lock the scheduler to protect BkUpAccess and ensure that the new time is write quickly